gymnasium/core.py

```diff
@@ -113,8 +115,8 @@ def reset(
         self,
         *,
         seed: Optional[int] = None,
-        options: Optional[dict] = None,
```

Comment:
I'd need to check, but I think that with removal of python 3.6 we can move (back) to using `dict[str, Any]` instead of `Dict[str, Any]`. It's not necessary in this PR, but might be a bit nicer long-term